### ISSUE TITLE
Provide an option (either dynamic or perhaps global) for xref's and include's to use the fully qualified format rather than local.

### ISSUE THREAD

**Issue by danhaywood** [asciidoctor/asciidoctor-intellij-plugin#418]
In particular, this is for Antora.

For example, suppose I have:

component-a/modules/foo/pages/x.adoc
component-a/modules/foo/pages/y.adoc
component-a/modules/bar/pages/p.adoc
component-b/modules/baz/pages/k.adoc

if I am working in an Antora module on x.adoc, and use xref:, then the options I will have are:

- y.adoc
- foo/y.adoc
- component-a/foo/y.adoc
- bar/p.adoc
- component-a/bar/p.adoc
- component-b/baz/k.adoc

This is somewhat bewildering.  I have a couple of ideas to restrict this.

a) perhaps specify globally whether to show the shorter forms or not in the list.  If set to component/module always, then the list would be just:

- component-a/foo/y.adoc
- component-a/bar/p.adoc
- component-b/baz/k.adoc

I am finding myself wanting to use this format everywhere.

b) perhaps allow the shorter form to be specified, but then provide intellisense for the longer form if "within" the xref link.

So, if I initially have xref:y.adoc[], but then decide I want the fully qualified form, then ctrl+space near to y.adoc should give me the other longer alternatives:

- foo/y.adoc
- component-a/foo/y.adoc

**Comment by ahus1** [comment]
@danhaywood - thanks for raising this issue. I wonder if some of the "/" should actually be ":" with the Antora syntax?

I'll invite some comments for this issue from the Antora users channel; I wonder what other users prefer. Once I know the preferences I can figure out what configuration options would be needed.

**Comment by ahus1** [comment]
Hi @danhaywood - when implementing the versions and after a helpful discussions with @djencks I realized that the local format behaves different from the fully qualified format: 
**If an xref specifies a component name (even if it specifies it own component name), it will link to the latest version** 

This is somewhat mentioned here (https://docs.antora.org/antora/2.2/asciidoc/page-to-page-xref/#different-component), although it doesn't mention what happens when you specify the current component's name for an xref. I gave it a try using the Antora 2.2 release and yes, any fully qualified xref without a version will link to the latest version, even if it is within the same component (thanks to David Antora example repo).

The docs for selecting the latest version are here: https://docs.antora.org/antora/2.2/component-versions/#version-sorting.

IMHO a user would expect to stay with the current version of the component unless they switch the component. Therefore using fully qualified links within a component would lead IMHO to a confusing result for the user.

Given this behavior I think local links should never have their local component name in the xref. 

The Apache Isis documentation has a lot of components. I wonder how you want to proceed with local links there.
I could offer an IntelliJ inspection + quickfix to remove component names from a component-local link.

**Comment by djencks** [comment]
A fully qualified xref would explicitly include the version, so it would not link to the latest version (unless your page was in the latest version).  Fully qualified is always the least ambiguous format.

**Comment by ahus1** [comment]
Thanks David, I've update my comment to talk about "xref with component name" instead.

**Comment by mojavelinux** [comment]
> If an xref specifies a component name (even if it specifies it own component name), it will link to the latest version

Indeed. The idea is that if you specify a component name, you're infering that you want to break out of the current context. That means the resolver won't consider the current version. Instead, it applies context-less resolution rules (same as if it the reference was coming from a different component). Otherwise, there would be no way to refer to a resource in the latest version of the current component.

If you want to specify the component name, but stay in the current version, you can use:

```
xref:{page-version}@component-name::index.adoc[]
```

(Though this assumes the page is in the `component-name` component).
